### PR TITLE
LibGfx: Reimplement from_hsla

### PR DESCRIPTION
**Pull request by glupi-borna**
- Hue now wraps properly when negative or larger than 360
- The hsl to rgb conversion now closely mirrors the code example from the spec.

This fixes a number of WPT tests in
/css/css-color/parsing/color-computed-hsl.html

### CODE REVIEW

File: Libraries/LibGfx/Color.h [hunk 1]
```diff
@@ -131,48 +131,36 @@ class Color {
...
+            float a = s * min(l, 1.0f - l);
+            g = l - a * max(-1.0f, min(min(k - 3.0f, 9.0f - k), 1.0f));
+        }
+
+        float b;
+        {
+            float k = fmodf(4.0f + h / 30.0f, 12.0f);
+            float a = s * min(l, 1.0f - l);
+            b = l - a * max(-1.0f, min(min(k - 3.0f, 9.0f - k), 1.0f));
+        }
```
Comment (gmta) [inline Libraries/LibGfx/Color.h:162]: Suggestion:

```cpp
auto to_rgb = [](float h, float offset) {
    float k = fmodf(offset + h / 30.0f, 12.0f);
    float a = s * min(l, 1.0f - l);
    return l - a * max(-1.0f, min(min(k - 3.0f, 9.0f - k), 1.0f));
};
auto r = to_rgb(h, 0.0f);
auto g = to_rgb(h, 8.0f);
auto b = to_rgb(h, 4.0f);
```

**Comment by LucasChollet** [comment]
It would also be nice to see some tests in `TestColor.cpp` for edge cases that are now working correctly.

**Comment by glupi-borna** [comment]
> It would also be nice to see some tests in `TestColor.cpp` for edge cases that are now working correctly.

Yep, went ahead and added tests for the WPT cases where we were off by 1 with the previous implementation, along with some tests to ensure that wrapping the hue works properly.

File: Libraries/LibGfx/Color.h [hunk 2]
```diff
@@ -131,48 +131,25 @@ class Color {
...
-        float r = hue_to_rgb(m1, m2, h + 1.0f / 3.0f);
-        // PUT hue.to.rgb(m1, m2, h    ) IN g
-        float g = hue_to_rgb(m1, m2, h);
-        // PUT hue.to.rgb(m1, m2, h-1/3) IN b
-        float b = hue_to_rgb(m1, m2, h - 1.0f / 3.0f);
+        float r = to_rgb(h, s, l, 0.0f);
+        float g = to_rgb(h, s, l, 8.0f);
+        float b = to_rgb(h, s, l, 4.0f);
+
         // RETURN (r, g, b)
```
Comment (LucasChollet) [inline Libraries/LibGfx/Color.h:153]: Don't hesitate to remove that useless comment as well (I should have caught that the first time, sorry).

**Review by LucasChollet — APPROVED** [review]
Except for a small nitpick, LGTM!

File: Libraries/LibGfx/Color.h (hunk 2 above)
Comment (glupi-borna) [reply Libraries/LibGfx/Color.h]: Got it, no worries! :)
